src/wp-includes/load.php

```diff
@@ -271,11 +271,15 @@ function wp_get_environment_type() {
  * The development mode affects how certain parts of the WordPress application behave, which is relevant when
  * developing for WordPress.
  *
- * Valid developer modes are 'core', 'plugin', 'theme', or an empty string to disable developer mode.
+ * Valid development modes are 'core', 'plugin', 'theme', 'all', or an empty string to disable development mode.
+ * 'all' is a special value to signify that all three development modes 'core', 'plugin', and 'theme' are enabled.
  *
  * Developer mode is considered separately from `WP_DEBUG` and {@see wp_get_environment_type()}. It does not affect
  * debugging output, but rather functional nuances in WordPress.
  *
+ * This function controls the currently set development mode value. To check for whether a specific development mode is
+ * enabled, use {@see wp_in_development_mode()}.
```

Comment:
```suggestion
 * enabled, use wp_in_development_mode().
```

The docs parser picks up function references automatically, it's only hooks that require an `@see` annotation.